src/main/java/io/seqera/tower/cli/commands/datasets/CreateCmd.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, Seqera Labs.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * This Source Code Form is "Incompatible With Secondary Licenses", as
+ * defined by the Mozilla Public License, v. 2.0.
+ */
+
+package io.seqera.tower.cli.commands.datasets;
+
+import io.seqera.tower.ApiException;
+import io.seqera.tower.cli.commands.global.WorkspaceOptionalOptions;
+import io.seqera.tower.cli.responses.Response;
+import io.seqera.tower.cli.responses.datasets.DatasetCreate;
+import io.seqera.tower.model.CreateDatasetRequest;
+import io.seqera.tower.model.CreateDatasetResponse;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+@CommandLine.Command(
+        name = "create",
+        description = "Create a workspace dataset."
+)
+public class CreateCmd extends AbstractDatasetsCmd{
+
+    @CommandLine.Option(names = {"-n", "--name"}, description = "Dataset name.", required = true)
+    public String name;
+
+    @CommandLine.Option(names = {"-d", "--description"}, description = "Dataset description.")
+    public String description;
+
+    @CommandLine.Option(names = {"--header"}, description = "Set first row as a header.")
+    public boolean header = false;
+
+    @CommandLine.Parameters(index = "0", paramLabel = "FILENAME", description = "Data file to upload.", arity = "1")
+    Path fileName = null;
+
+    @CommandLine.Mixin
+    public WorkspaceOptionalOptions workspace;
```

Comment:
Workspace must be required because datasets it is not available at personal workspace.

Reply:
I am assuming here that the user will pull from the `env` variable for workspace settings. If I force the workspace option on dataset commands, the `env` will be ignored. On the other hand, not having such a variable will lead to an error. It could be confusing. A mandatory default profile system is becoming more necessary as we evolve.

Reply:
If the variable is available then the picocli `required` flag it is not going to give error and use it. And if the user do not give a `-w` or sets the `env` variable then the command has to fail with a required `workspace` error.